relate/src/main/scala/com/lucidchart/relate/RowIterator.scala

```diff
@@ -32,9 +32,9 @@ private[relate] class RowIterator[A](parser: SqlRow => A, stmt: PreparedStatemen
    *   the parsed record
    */
   override def next(): A = {
-    val ret = parser(result.asRow)
+    val ret = parser(SqlRow(result.resultSet))
     if (_hasNext) {
-      _hasNext = result.next()
+      _hasNext = result.resultSet.next()
     }
 
     // if we've iterated through the whole thing, close resources
```

Comment:
Something I noticed, when looking over this, is that if we stop iterating over the iterator, then we won't close the resultSet.

I wonder if we should also replace `asIterator` with `withIterator` that calls a function with the Iterator, and ensure that it is closed at the end.

Reply:
:upvote:

Reply:
I think `asIterator` intentionally skips the auto-closing stuff to give the user more control. I looked into this but wasn't able to quickly decide a good way to structure things as there are a lot of different pieces involved (e.g. this is the only place that uses [StreamedStatementPreparer](https://github.com/lucidsoftware/relate/blob/master/relate/src/main/scala/com/lucidchart/relate/StatementPreparer.scala#L148)). I think it could be good to take a more comprehensive look at cleaning up/improving relate, but my main goal here is just to try to track down some data service issues, so I'm going to say this is out of scope